Assets/Scripts/Battle/UIController.cs

```diff
@@ -581,6 +581,13 @@ public enum UIState {
             throw new CYFException("An error occured while trying to enter the state \"" + state + "\":\n\n" + ex.Message + "\n\nTraceback (for devs):\n" + ex);
         }
     }
+	
+	public static void CreateNewUIState(string name) {
+		if (instance.UIStates.Contains(name))
+			throw new CYFException("The state \"" + name + "\" already exists, but you tried to create it again.");
```

Comment:
wordy, the second part isn't needed. `The state "name" already exists.`

Reply:
Got it.

Reply:
Also, as far as Rhen told me, the script gets re-loaded every time, so there is no need . . .

I can try testing it, though.

Reply:
![image](https://user-images.githubusercontent.com/48065803/122683916-ef3a1880-d20a-11eb-9b23-2d15145c8ec1.png)
This is the code I tried testing if the states get cleared between battles.
I opened and closed the testing mod multiple times within the same session, and the messages were always the same:

![image](https://user-images.githubusercontent.com/48065803/122683940-23153e00-d20b-11eb-9017-3d0754add997.png)
(The other debug messages between the two booleans are a result of switching to the "NONE1" state on the second time.)